internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -377,6 +393,55 @@ func (k *KafkaManager) updateClusterStatusCapacityMaxMetric(c services.KafkaStre
 	metrics.UpdateClusterStatusCapacityMaxCount(c.CloudProvider, c.Region, c.InstanceType, c.ClusterId, float64(c.MaxUnits))
 }
 
+// TODO: this is a temporary reconciler added to value the `actual_kafka_billing_model` and `desired_kafka_billing_model` for
+// kafkas where billing_model has always been empty. See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1115#discussion_r902415419 for details
+func (k *KafkaManager) reconcileBillingModel(kafkas dbapi.KafkaList) serviceErr.ErrorList {
+	var svcErrors serviceErr.ErrorList
+
+	for _, kafka := range kafkas {
+		if kafka.ActualKafkaBillingModel != "" {
```

Comment:
we could also ignore Kafkas in other deprovision or deleting states

Reply:
True, but I don't think it is a usefult optimisation for this temporary code.

Reply:
It could happen that the first time this reconciler runs, if the problematic Kafkas are in deleting state, we could run onto a situation where a bunch of errors are sent to reported https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f4bbc3a5b5d0048b5444bf5741fe8f98771459ca/internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go#L422-L428 

The check could save us (the Eng on primary) from investigation time.

Reply:
> It could happen that the first time this reconciler runs, if the problematic Kafkas are in deleting state, we could run onto a situation where a bunch of errors are sent to reported

Why? Even if deleted the subscription is still there in `deleted` status.

Reply:
Okay. I was not aware of that, I thought delete completely removes the subscription from AMS. Thanks for the clarification. In that case the errors won't happen.